packages/genesys-spark-components/src/components/beta/gux-avatar-group/gux-avatar-groups.tsx

```diff
@@ -0,0 +1,229 @@
+      this.overflowAvatars = Array.from(
+        this.avatarList.slice(maxAvatarsToDisplay)
+      );
+
+      if (this.overflowAvatars.length > 0) {
+        this.buildOverflowList();
+      }
+
+      this.overflowCount = this.countAvatars - maxAvatarsToDisplay;
+    });
+  }
+
+  private getMaxAvatarsToDisplay(containerWidth: number): number {
+    if (containerWidth <= 59) {
+      return 0;
+    } else if (containerWidth <= 83) {
+      return 1;
+    } else if (containerWidth <= 107) {
+      return 2;
+    } else if (containerWidth <= 131) {
+      return 3;
+    } else if (containerWidth <= 159) {
+      return 4;
+    } else if (containerWidth <= 183) {
+      return 5;
+    } else if (containerWidth <= 207) {
+      return 6;
+    } else {
+      return 7;
+    }
+  }
+
+  async componentWillLoad(): Promise<void> {
+    this.avatarList = Array.from(
+      this.root.children
+    ) as HTMLGuxAvatarFocusableBetaElement[];
+    trackComponent(this.root);
+  }
+
+  componentDidLoad() {
+    if (!this.resizeObserver && window.ResizeObserver) {
+      this.resizeObserver = new ResizeObserver(() =>
+        this.checkGroupContainerWidthForLayout()
+      );
+    }
+
+    if (this.resizeObserver) {
+      this.resizeObserver.observe(
+        this.root.shadowRoot.querySelector('.gux-avatar-group')
+      );
+    }
+
+    afterNextRenderTimeout(() => {
+      this.checkGroupContainerWidthForLayout();
+    }, 500);
+  }
+
+  disconnectedCallback() {
+    if (this.resizeObserver) {
+      this.resizeObserver.unobserve(
+        this.root.shadowRoot.querySelector('.gux-avatar-group')
+      );
+    }
+
+    if (this.listItemClickListeners.size > 0) {
+      this.removeAllListeners();
+    }
+  }
+
+  private slotChanged(): void {
+    this.avatarList = Array.from(
+      this.root.children
+    ) as HTMLGuxAvatarFocusableBetaElement[];
+
+    this.avatarList.forEach((avatar, index) => {
+      const avatarComp = avatar.querySelector(
+        'gux-avatar-beta'
+      ) as HTMLGuxAvatarBetaElement;
+
+      if (avatarComp) {
+        avatarComp.accent = this.getAccent(index) as GuxAvatarAccent;
+      }
+    });
+  }
+
+  private getAccent(index): string {
+    return ((index % 12) + 1).toString();
+  }
+
+  private showOverflowIndicator(): JSX.Element | null {
+    if (this.overflowCount > 0) {
+      return (
+        <gux-avatar-overflow-beta count={this.overflowCount}>
+          <gux-list></gux-list>
+        </gux-avatar-overflow-beta>
+      ) as JSX.Element;
+    } else {
+      return null;
+    }
+  }
+
+  private buildOverflowList(): void {
+    const list = this.root.shadowRoot.querySelector('gux-list');
+    if (list) {
+      list.innerHTML = '';
+      this.overflowAvatars.map(focusableAvatar =>
+        this.addListItemForAvatar(focusableAvatar)
+      );
+    }
+  }
+
+  private addListItemForAvatar(
+    focusableAvatar: HTMLGuxAvatarFocusableBetaElement
+  ): void {
+    const listItem = document.createElement('gux-list-item');
```

Comment:
Instead of manually creating elements could we have a function that renders a list of items based on the value of a javascript variable value?

Reply:
Also could we document more of the business logic at the top of this function for my clarity of the functionality?